src/collectArgs.js

```diff
@@ -0,0 +1,75 @@
+/**
+* A helper method interpreting a given function definition
+* following the function spec from TODO:link-to-func-specification
+*/
+export default async function collectArgs (func, call, hooks) {
+  // HACK: just use the first method
+  // TODO: why would there be multiple ones?
```

Comment:
There might be multiple methods for different arguments types/argument signatures ie. "multimethods" as in Julia etc